src/movies.js

```diff
@@ -1,32 +1,112 @@
-// The `movies` array from the file `src/data.js`.
-console.log('movies: ', movies);
 
+const movies = require('../src/data.js');
+// turnHoursToMinutes(movies2);
+//console.log( bestYearAvg(movies2));
 
-// Iteration 1: All directors? - Get the array of all directors.
-// _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
-// How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(arr) 
+{
+  const AllDirectors = arr.map(function(element)
+  {
+  return element.director
+  }
+  ) 
+  return AllDirectors;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(arr) {
+  const Dramas =  arr.filter(movie => {
+    return movie.director === 'Steven Spielberg'&& movie.genre.includes
+    ("Drama");
+  })
+  
+    return Dramas.length;
+  }
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(arr) {
+  if (arr.length === 0){return 0;};
+  const scores = arr.reduce(function(accumulator, currentValue)
+  {
+    if(!currentValue.score){
+      currentValue.score = 0;
+    };
+    return accumulator + currentValue.score
+  },0);
+  const scoreAverage = scores / arr.length;
+  return Number(scoreAverage.toFixed(2));
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore(arr) {
+  const dramaAvg = arr.filter(movie => {
+    return movie.genre.includes('Drama');
+  });
+  return scoresAverage(dramaAvg);
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+//a.title.toLowerCase().localeCompare(b.title.toLowerCase())
+function orderByYear(arr) {
+  const sortMoviesByYear =[arr];
```

Comment:
this add the original array into a new one, so you basically end up having
```javascript
[
  [
    {
      "title":"Blade Runner"
    },
    // all the other movies here
  ]
]
```
in order to succeed in what you had to do, you either needed it to copy in some way, or use destructuring (we looked at it briefely in class, it's the three dots `...`